OneCommunityGlobal/HGNRest
# Tim: BM Dashboard - Add Equipment API

# Description
Implements Phase 2 WBS # 9.2.8, which is an API for the Add Equipment Type form. The API receives a POST request and checks whether the type already exists in the buildingInventoryTypes collection. If it does, a 409 error is returned. If not, a new document is added to buildingInventoryType with category "Equipment".

## Related PRS (if any):
To test this backend PR you need to checkout the #XXX frontend PR.

## Main changes explained:
- Adds a new API that receives a POST request to add a new "Equipment" type to buildingInventoryTypes collection
- The API first checks whether the type already exists in the buildingInventoryTypes collection.
   - If it does, a 409 error is returned.
   - If not, a new document is added to buildingInventoryType with category "Equipment".
- Updates **buildingInventoryType** model with Mongoose discriminator for each type category


## How to test:
1. `git pull` on the development branch to make sure you are up to date
2. `git checkout Tim_add_equip_api`
3. `nvm i`, then `npm i`
4. `npm run build`, then `npm run start`
5. follow frontend testing instructions
